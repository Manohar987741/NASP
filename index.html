

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Welcome to Northern Arizona SNP Pipeline’s documentation! &mdash; Northern Arizona SNP Pipeline 0.9.8 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  
    <link rel="top" title="Northern Arizona SNP Pipeline 0.9.8 documentation" href="#"/>
        <link rel="next" title="The Northern Arizona SNP Pipeline (NASP)" href="readme.html"/> 

  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="#" class="fa fa-home"> Northern Arizona SNP Pipeline</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
        
            <ul>
<li class="toctree-l1"><a class="reference internal" href="readme.html">The Northern Arizona SNP Pipeline (NASP)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="readme.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#getting-started">Getting Started</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#repeating-analysis">Repeating Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#license">License</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#contact">Contact</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#install">Install</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#update">Update</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#uninstall">Uninstall</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#required-dependencies">Required Dependencies</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#optional-dependencies">Optional Dependencies</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a><ul>
<li class="toctree-l2"><a class="reference internal" href="usage.html#nasp">nasp</a></li>
<li class="toctree-l2"><a class="reference internal" href="usage.html#format-fasta">format_fasta</a></li>
<li class="toctree-l2"><a class="reference internal" href="usage.html#find-duplicates">find_duplicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="usage.html#vcf-to-matrix">vcf_to_matrix</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="scripts.html">Post-Analysis Scripts</a><ul>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#filter-matrix-by-coord-py">filter_matrix_by_coord.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#filter-matrix-by-genome-py">filter_matrix_by_genome.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#matrix-to-fasta-py">matrix_to_fasta.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#report-single-snps-single-isolate-py">report_single_snps_single_isolate.py</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#types-of-contributions">Types of Contributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#get-started">Get Started!</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#pull-request-guidelines">Pull Request Guidelines</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#tips">Tips</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="authors.html">Credits</a><ul>
<li class="toctree-l2"><a class="reference internal" href="authors.html#development-lead">Development Lead</a></li>
<li class="toctree-l2"><a class="reference internal" href="authors.html#contributors">Contributors</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="history.html">History</a></li>
<li class="toctree-l1"><a class="reference internal" href="history.html#id1">0.9.6 (2014-06-05)</a></li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="#">Northern Arizona SNP Pipeline</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="#">Docs</a> &raquo;</li>
      
    <li>Welcome to Northern Arizona SNP Pipeline&#8217;s documentation!</li>
      <li class="wy-breadcrumbs-aside">
        
          <a href="_sources/index.txt" rel="nofollow"> View page source</a>
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            
  <div class="section" id="welcome-to-northern-arizona-snp-pipeline-s-documentation">
<h1>Welcome to Northern Arizona SNP Pipeline&#8217;s documentation!<a class="headerlink" href="#welcome-to-northern-arizona-snp-pipeline-s-documentation" title="Permalink to this headline">¶</a></h1>
<p>Contents:</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="readme.html">The Northern Arizona SNP Pipeline (NASP)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="readme.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#getting-started">Getting Started</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#repeating-analysis">Repeating Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#license">License</a></li>
<li class="toctree-l2"><a class="reference internal" href="readme.html#contact">Contact</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#install">Install</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#update">Update</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#uninstall">Uninstall</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#required-dependencies">Required Dependencies</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#optional-dependencies">Optional Dependencies</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a><ul>
<li class="toctree-l2"><a class="reference internal" href="usage.html#nasp">nasp</a></li>
<li class="toctree-l2"><a class="reference internal" href="usage.html#format-fasta">format_fasta</a></li>
<li class="toctree-l2"><a class="reference internal" href="usage.html#find-duplicates">find_duplicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="usage.html#vcf-to-matrix">vcf_to_matrix</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="scripts.html">Post-Analysis Scripts</a><ul>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#filter-matrix-by-coord-py">filter_matrix_by_coord.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#filter-matrix-by-genome-py">filter_matrix_by_genome.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#matrix-to-fasta-py">matrix_to_fasta.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="scripts.html#report-single-snps-single-isolate-py">report_single_snps_single_isolate.py</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#types-of-contributions">Types of Contributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#get-started">Get Started!</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#pull-request-guidelines">Pull Request Guidelines</a></li>
<li class="toctree-l2"><a class="reference internal" href="contributing.html#tips">Tips</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="authors.html">Credits</a><ul>
<li class="toctree-l2"><a class="reference internal" href="authors.html#development-lead">Development Lead</a></li>
<li class="toctree-l2"><a class="reference internal" href="authors.html#contributors">Contributors</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="history.html">History</a></li>
<li class="toctree-l1"><a class="reference internal" href="history.html#id1">0.9.6 (2014-06-05)</a></li>
</ul>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference internal" href="genindex.html"><em>Index</em></a></li>
<li><a class="reference internal" href="py-modindex.html"><em>Module Index</em></a></li>
<li><a class="reference internal" href="search.html"><em>Search Page</em></a></li>
</ul>
<a href="https://github.com/TGenNorth/NASP"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://camo.githubusercontent.com/e7bbb0521b397edbd5fe43e7f760759336b5e05f/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f6769746875622f726962626f6e732f666f726b6d655f72696768745f677265656e5f3030373230302e706e67" alt="Fork me on GitHub" data-canonical-src="https://s3.amazonaws.com/github/ribbons/forkme_right_green_007200.png"></a></div>


          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="readme.html" class="btn btn-neutral float-right" title="The Northern Arizona SNP Pipeline (NASP)"/>Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2014, Darrin Lemmer.
    </p>
  </div>

  <a href="https://github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.9.8',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>