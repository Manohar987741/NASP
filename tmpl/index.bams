#!/bin/bash
source ${workdir}/job_scripts/common.vars

trap 'eval "$snpcaller_release_cmd"' EXIT

in_bam=${bams[$index]}
bam_basename=${bam##*/}
bam_filename=${bam_basename%.*}
out_bam=${workdir}/bams/${bam_basename}

samtools=$(get_program_path "samtools")

# GATK requires the ReadGroup header. If it is not present, assign as default value.
if $samtools view -H $bam | grep "^@RG"; then
    ln -s $bam ${workdir}/bams
else
    picard=$(get_program_path "picard")
    # Handle cases where picard is either a jar file or a wrapper script
    [[ "$picard" = *.jar ]] && picard="java -jar $picard" 
    # TODO: Exit if $picard is not picard.jar
    $picard AddOrReplaceReadGroups \
        INPUT="$in_bam" \
        OUTPUT="$out_bam" \
        SORT_ORDER="coordinate" \
        ID="1" \
        SM="$bam_filename" \
        LB="$bam_filename" \
        PL="illumina" \
        PU="1"
fi

$samtools index ${out_bam}
