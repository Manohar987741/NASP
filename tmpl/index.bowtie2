#!/bin/bash
source ${workdir}/job_scripts/common.vars

# bowtie2-build will create ${reference%.*}.{1.bt2,2.bt2,3.bt2,4.bt2,rev.1.bt2,rev.2.bt2} in the ${reference} directory
$(get_program_path "bowtie2-build") ${reference} ${reference%.*}
