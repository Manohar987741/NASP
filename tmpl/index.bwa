#!/bin/bash
source ${workdir}/job_scripts/common.vars

$(get_program_path "bwa") index "$reference"
