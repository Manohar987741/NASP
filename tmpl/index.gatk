#!/bin/bash
source ${workdir}/job_scripts/common.vars

picard=$(get_program_path "picard")
# Before picard v1.124 the picard subcommands were discrete jar files.
if [[ "${picard##*/}" = "CreateSequenceDictionary.jar" ]]; then
    CreateSequenceDictionary="$picard"
else
    CreateSequenceDictionary="$picard CreateSequenceDictionary"
fi

# Creates ${reference%.*}.dict in ${reference} directory
$(get_program_path "java") \
      -Xmx2G \
      -jar ${CreateSequenceDictionary:?} \
      R="$reference" \
      O="${reference%.*}.dict"

# Creates ${reference}.fai in ${reference} directory
$(get_program_path "samtools") faidx "$reference"
