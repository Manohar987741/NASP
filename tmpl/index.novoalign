#!/bin/bash
source ${workdir}/job_scripts/common.vars

$(get_program_path "novoindex") ${reference}.idx ${reference}
