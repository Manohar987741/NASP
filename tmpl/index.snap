#!/bin/bash
source ${workdir}/job_scripts/common.vars

$(get_program_path "snap") index ${reference} ${reference%.*}
